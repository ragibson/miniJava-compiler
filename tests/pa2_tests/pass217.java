// PA2 pass stmt
class A {
    void f(int a, int b){
        if (a < b) b = a;
    }
}
