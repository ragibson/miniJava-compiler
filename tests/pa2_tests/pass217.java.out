********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "f" methodname
  .   .   ParameterDeclList [2]
  .   .   . ParameterDecl
  .   .   .   INT BaseType
  .   .   .   "a"parametername 
  .   .   . ParameterDecl
  .   .   .   INT BaseType
  .   .   .   "b"parametername 
  .   .   StmtList [1]
  .   .   . IfStmt
  .   .   .   BinaryExpr
  .   .   .     "<" Operator
  .   .   .       RefExpr
  .   .   .         IdRef
  .   .   .           "a" Identifier
  .   .   .       RefExpr
  .   .   .         IdRef
  .   .   .           "b" Identifier
  .   .   .   AssignStmt
  .   .   .     IdRef
  .   .   .       "b" Identifier
  .   .   .     RefExpr
  .   .   .       IdRef
  .   .   .         "a" Identifier
=============================================
